test/test_loaded_meshes.cpp

```diff
@@ -0,0 +1,254 @@
+//    loaded_convex_meshes.push_back(new bodies::ConvexMesh(loaded_meshes.back()));
+
+    // CYLINDER
+    shapes::Cylinder cylinder(0.5, 1.0);
+    shape_meshes.push_back(shapes::createMeshFromShape(&cylinder));
+    loaded_meshes.push_back(shapes::createMeshFromResource(
+        "file://" + (boost::filesystem::path(TEST_RESOURCES_DIR) / "/cylinder.stl").string()));
+
+    shape_convex_meshes.push_back(new bodies::ConvexMesh(shape_meshes.back()));
+    loaded_convex_meshes.push_back(new bodies::ConvexMesh(loaded_meshes.back()));
+
+    // CONE
+    shapes::Cone cone(0.5, 1.0);
+    shape_meshes.push_back(shapes::createMeshFromShape(&cone));
+    loaded_meshes.push_back(shapes::createMeshFromResource(
+        "file://" + (boost::filesystem::path(TEST_RESOURCES_DIR) / "/cone.stl").string()));
+
+    shape_convex_meshes.push_back(new bodies::ConvexMesh(shape_meshes.back()));
+    loaded_convex_meshes.push_back(new bodies::ConvexMesh(loaded_meshes.back()));
+
+    // BOX
+    shapes::Box box(1.0, 1.0, 1.0);
+    shape_meshes.push_back(shapes::createMeshFromShape(&box));
+    loaded_meshes.push_back(shapes::createMeshFromResource(
+        "file://" + (boost::filesystem::path(TEST_RESOURCES_DIR) / "/cube.stl").string()));
+
+    shape_convex_meshes.push_back(new bodies::ConvexMesh(shape_meshes.back()));
+    loaded_convex_meshes.push_back(new bodies::ConvexMesh(loaded_meshes.back()));
+  }
+
+  void TearDown()
+  {
+    for (int i = 0; i < shape_meshes.size(); ++i)
+    {
+      delete shape_meshes[i];
+      delete loaded_meshes[i];
+
+      delete shape_convex_meshes[i];
+      delete loaded_convex_meshes[i];
+    }
+  }
+
+  ~CompareMeshVsPrimitive()
+  {
+  }
+
+protected:
+  random_numbers::RandomNumberGenerator rng;
+
+  std::vector<shapes::Mesh*> shape_meshes;
+  std::vector<shapes::Mesh*> loaded_meshes;
+
+  std::vector<bodies::Body*> shape_convex_meshes;
+  std::vector<bodies::Body*> loaded_convex_meshes;
+};
+
+TEST_F(CompareMeshVsPrimitive, ContainsPoint)
+{
+  // Any point inside a mesh must be inside the other too
+  for (int i = 0; i < shape_meshes.size(); ++i)
+  {
+    shapes::Mesh *shape_ms = shape_meshes[i];
+    shapes::Mesh *loaded_ms = loaded_meshes[i];
+
+    bodies::Body *shape_cms = shape_convex_meshes[i];
+    bodies::Body *loaded_cms = loaded_convex_meshes[i];
+
+    Eigen::Vector3d p;
+    bool found = false;
+    for (int i = 0; i < 100; ++i)
+    {
+      if ((shape_cms->samplePointInside(rng, 10000, p)) ||
+          (loaded_cms->samplePointInside(rng, 10000, p)))
+      {
+        found = true;
+        EXPECT_EQ(shape_cms->containsPoint(p), loaded_cms->containsPoint(p));
+      }
+    }
+    EXPECT_TRUE(found) << "No point inside the meshes was found (very unlikely)";
+  }
+}
+
+TEST_F(CompareMeshVsPrimitive, IntersectsRay)
+{
+  // Random rays must intersect both meshes nearly at the same points
+
+  SUCCEED() << "Test disabled by now, as it fails in around 1% cases";
```

Comment:
Maybe just choose some "nice" rays testing corner cases, instead of just random rays.
However, random rays should always work, if both meshes are exactly the same and with the same rotation. I wonder if one is slightly rotated relative to the other... this, together with conic shapes discretization can explain the discrepancies. I guess increasing the resolution used to create the conics will reduce the 1% of failures, but the value is hardcoded on shapes constructors